### ISSUE TITLE
Scaling out does not seem to trigger

### ISSUE THREAD

**Issue by khebul** [Azure/azure-functions-dotnet-worker#649]
Hi, 

A few weeks back we have migrated .NET Core 3.1 Azure Function App to dotnet-isolated worker, running on Consumption plan. During low load periods, everything runs as expected. However, during peak periods there are problems.

In those periods HTTP triggers constantly time out, I can see messages like this in the logs: ```Timeout value of 00:05:00 was exceeded by function: <FunctionName> ```, ```A function timeout has occurred. Restarting worker process execu```, ```Restart of language worker process(es) completed.```. Diagnostic tools also give me information about only one function app process running, and it does not seem to start any new processes. CPU load during this time stays fairly high as well: 70-100%.

When I try the same test with .NET Core 3.1, it passes with no significant issues. I see that Azure starts new processes to handle the increased load and kills them when they are no longer needed.

Is scaling out supposed to be different for dotnet-isolated worker? How can I avoid this issue with dotnet-isolated worker?

**Comment by khebul** [comment]
Hey, I see that this ticket is not assigned to anyone, and there have been no updates so far. Is there anything I can do do give you more information about the issues?

**Comment by fabiocav** [comment]
@khebul the application name would be helpful to identify what is causing the issue and the scaling behavior. 

For GitHub issues, we typically triage and review what is opened in order, but if this is a production impacting issue, engaging support might be a better option.

Thanks!

**Comment by snowke** [comment]
I'm curious about this as well.  I see a similar situation where I can not trigger scaling.

Setup
We have an Azure Function App written in .NET 5.   The app is running on a windows machine in a consumption plan in Azure GCC. The app consists of 3 HTTP Triggered functions.  Two of the functions are high volume, low resource methods.  The 3rd function contains a resource-intensive operation where we digital sign data with a certificate.

Load Test
In parallel for each of the respective functions, we gradually build up to 150+ virtual users over the course of 60 minutes.

Result
We push the function app to the point of timeouts, too many requests, going over 100% CPU, etc.  But the function app never scales above one server.  See below for an example of viewing the results in Application Insights as we run K6 load tests.

![image](https://user-images.githubusercontent.com/6555327/140195451-a75a27b6-6ad3-423b-afe5-87de00547f13.png)
